File: 프로그래머스/unrated/120808. 분수의 덧셈/분수의 덧셈.py
def solution(numer1, denom1, numer2, denom2):
    numer3=(numer1*denom2) + (numer2*denom1)
    denom3=(denom2*denom1)
    for i in range (denom3, 1, -1):
        if numer3%i == 0 and denom3%i == 0:
            numer3 = numer3//i
            denom3 = denom3//i
    answer = [numer3, denom3]
    return answer

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = {};
        int numer = numer1 * denom2 + numer2 * denom1;
        int denom = denom1 * denom2;
        
        int g = gcd(numer, denom);
        
        return new int[] { numer / g, denom /g };
    }
    
    private int gcd(int a, int b) {
    // 유클리드 호제법
    while (b != 0) {
        int tmp = a % b;
        a = b;
        b = tmp;
    }
    return a;
    }
}